library/lua/repeat-util.lua

```diff
@@ -17,7 +17,9 @@ function cancel(name)
  if not repeating[name] then
   return false
  end
- dfhack.timeout_active(repeating[name],nil)
+ if repeating[name] ~= -1 then
+  dfhack.timeout_active(repeating[name],nil)
```

Comment:
I'm using -1 as a placeholder value for repeating tasks that are running for the first time and haven't been re-scheduled yet.